File: datastruct/stack/LinkedListStack.java
public class LinkedListStack<E> implements Stack<E> {

    private LinkedList<E> list;

    public LinkedListStack(){
        list = new LinkedList<>();
    }

    @Override
    public int getSize() {
        return list.getSize();
    }

    @Override
    public boolean isEmpty() {
        return list.isEmpty();
    }

    @Override
    public E pop() {
        return list.removeFirst();
    }

    @Override
    public void push(E ele) {
        list.addToFirst(ele);
    }

    @Override
    public E peek() {
        return list.getFirst();
    }

    @Override
    public String toString() {
        StringBuilder str = new StringBuilder();
        str.append("Stack: top ");
        str.append(list);
        return str.toString();
    }
}

File: datastruct/stack/LinkedListStack.py
from LinkedList import LinkedList
from Stack import Stack

class LinkedListStack(Stack):
    def __init__(self):
        self.__list = LinkedList()
    
    def get_size(self):
        return self.__list.get_size()
    
    def is_empty(self):
        return self.__list.is_empty()
    
    def push(self, ele):
        self.__list.add_to_first(ele)
    
    def pop(self):
        return self.__list.remove_first()
    
    def peek(self):
        return self.__list.get_first()
    
    def __str__(self):
        string = 'Stack: top '
        string += str(self.__list)
        return string
